src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -0,0 +1,271 @@
+{
+    /// <summary>
+    /// Contains tests that ensure the correctness of any class that implements the nongeneric
+    /// ICollection interface
+    /// </summary>
+    public abstract class ICollection_NonGeneric_Tests : IEnumerable_NonGeneric_Tests
+    {
+        #region Helper methods
+
+        /// <summary>
+        /// Creates an instance of an ICollection that can be used for testing.
+        /// </summary>
+        /// <returns>An instance of an ICollection that can be used for testing.</returns>
+        protected abstract ICollection NonGenericICollectionFactory();
+
+        /// <summary>
+        /// Creates an instance of an ICollection that can be used for testing.
+        /// </summary>
+        /// <param name="count">The number of unique items that the returned ICollection contains.</param>
+        /// <returns>An instance of an ICollection that can be used for testing.</returns>
+        protected virtual ICollection NonGenericICollectionFactory(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory();
+            AddToCollection(collection, count);
+            return collection;
+        }
+
+        protected virtual bool DuplicateValuesAllowed { get { return true; } }
+        protected virtual bool IsReadOnly { get { return false; } }
+        protected virtual bool NullAllowed { get { return true; } }
+        protected virtual bool ExpectedIsSynchronized { get { return false; } }
+        protected virtual IEnumerable<object> InvalidValues { get { return new object[] { }; } }
+        protected abstract void AddToCollection(ICollection collection, int numberOfItemsToAdd);
+
+        #endregion
+
+        #region IEnumerable Helper Methods
+
+        protected override int WaysToModify { get { return 0; } }
+
+        protected override IEnumerable NonGenericIEnumerableFactory(int count)
+        {
+            return NonGenericICollectionFactory(count);
+        }
+
+        #endregion
+
+        #region Count
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_Count_Validity(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Assert.Equal(count, collection.Count);
+        }
+
+        #endregion
+
+        #region IsSynchronized
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_IsSynchronized(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Assert.Equal(ExpectedIsSynchronized, collection.IsSynchronized);
+        }
+
+        #endregion
+
+        #region SyncRoot
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_SyncRoot(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            var root = collection.SyncRoot; // doesnt throw
```

Comment:
I see now you have both of those covered in tests below... what does this test cover that those don't?